app/models/symphony_curr_loc_request.rb

```diff
@@ -0,0 +1,50 @@
+require 'json'
+
+# API for retrieving the current location from a barcode via Symphony Web Services
+class SymphonyCurrLocRequest
+  include ActiveModel::Model # allows initialization with a hash of attributes
+
+  attr_accessor :barcode
+
+  def current_location
+    return '' if json.empty?
+    json['fields']['currentLocation']['key']
+  rescue
+    return ''
+  end
```

Comment:
Not sure this should silenty return empty string if bad or missing json ...  kinda followed the pattern from `symphony_user_email_requests`